File: hw/rojas/u1/HW11ChickenCoops/ChickenCoop.js
class ChickenCoop {
  constructor(id, name) {
    this.id = id; this.name = name;
    this.chickens = []; this.eggs = []; this.poops = [];
  }
  addChicken(chicken) { this.chickens.push(chicken); }
  collectEgg(egg) { if (egg) this.eggs.push(egg); }
  collectPoop(poop) { if (poop) this.poops.push(poop); }
  showInfo() {
    console.log(`\nCoop ${this.id} - ${this.name}`);
    console.log(`  Chickens: ${this.chickens.length}`);
    this.chickens.forEach(c => console.log("   • " + c.toString()));
    console.log(`  Eggs: ${this.eggs.length}, Poops: ${this.poops.length}`);
  }
}
export { ChickenCoop };

File: hw/rojas/u1/HW11ChickenCoops/chicken_coop.py
class ChickenCoop:
    def __init__(self, id, name):
        self.id = id
        self.name = name
        self.chickens = []
        self.eggs = []
        self.poops = []

    def add_chicken(self, chicken):
        self.chickens.append(chicken)

    def collect_egg(self, egg):
        if egg:
            self.eggs.append(egg)

    def collect_poop(self, poop):
        if poop:
            self.poops.append(poop)

    def show_info(self):
        print(f"\n Coop ID: {self.id} | Name: {self.name}")
        print(f"Chickens ({len(self.chickens)} total):")
        for chicken01 in self.chickens:
            print(f" → {chicken01}")
        print(f"Eggs collected: {len(self.eggs)}")
        print(f"Poops collected: {len(self.poops)}")